RestartManager.js

```diff
@@ -0,0 +1,33 @@
+let NativeCodePush = require("react-native").NativeModules.CodePush;
+
+const RestartManager = (() => {
+    let _allowed = true;
+
+    function restartApp(onlyIfUpdateIsPending = false) {
+        if (_allowed) {
+            NativeCodePush.restartApp(onlyIfUpdateIsPending);
+        }
+    }
+
+    function allow() {
```

Comment:
i don't think it's necessary to track whether or not a restart was requested. i think what needs to happen is if an update is pending and its restart mode is IMMEDIATE then it needs to restart the app here, and otherwise shouldn't. tracking whether a restart was requested gets close to that behavior, but i'd be worried that it may lead to issues with situations like the following:
```
- restarts are disallowed
- IMMEDIATE update gets downloaded, wants to restart but is rejected
- ON_NEXT_RESUME update gets downloaded, doesn't try to restart
- restarts are allowed again
- app restarts because a restart was rejected while it was disallowed
```
this shouldn't happen because the pending update is a ON_NEXT_RESUME update